ui/src/app/kvm/components/KVMHeaderForms/AddKVM/AddLxd/AddLxd.tsx

```diff
@@ -78,14 +77,11 @@ export const AddLxd = ({
     }
   }, [shouldGoToProjectStep]);
 
+  // We run the cleanup function here rather than in each form component
+  // because we want the errors to be able to persist across forms.
   useEffect(() => {
     return () => {
-      // We run the cleanup function here rather than in each form component
-      // because we want the errors to be able to persist across forms.
       dispatch(podActions.cleanup());
-      // The generated certificate is also cleared as we only store one in state
-      // at a time. This will prepare the form for the next added VM host.
-      dispatch(generalActions.clearGeneratedCertificate());
```

Comment:
This has been moved to `AuthenticationForm`'s unmount function, otherwise if an error occurred on the authentication step the `useEffect()`s would cause the user to infinitely bounce between the "credential" and "authentication" steps.